index.js

```diff
@@ -28,6 +28,23 @@ function requestAsEventEmitter(opts) {
 	var get = function (opts) {
 		var fn = opts.protocol === 'https:' ? https : http;
 
+		// check for unix domain socket
+		if (opts.hostname === 'unix') {
```

Comment:
Is this check belongs here? Why not in `normalizeAgruments` (redirect can return unix socket in headers?)?